src/idl_gen_ts.cpp

```diff
@@ -219,68 +223,91 @@ class TsGenerator : public BaseGenerator {
-          // TODO: Make it so we don't generate a self-import; this will also
-          // require modifying AddImport to ensure that we don't use
-          // namespace-prefixed names anywhere...
-          std::string file = it.first;
-          if (file.empty()) { continue; }
-          std::string noext = flatbuffers::StripExtension(file);
-          std::string basename = flatbuffers::StripPath(noext);
-          std::string include_file = GeneratedFileName(
-              parser_.opts.include_prefix,
-              parser_.opts.keep_prefix ? noext : basename, parser_.opts);
-          // TODO: what is the right behavior when different include flags are
-          // specified here? Should we always be adding the "./" for a relative
-          // path or turn it off if --include-prefix is specified, or something
-          // else?
-          std::string include_name =
-              "./" + flatbuffers::StripExtension(include_file);
-          code += "import {";
-          for (const auto &pair : it.second) {
-            code += namer_.EscapeKeyword(pair.first) + " as " +
-                    namer_.EscapeKeyword(pair.second) + ", ";
-          }
-          code.resize(code.size() - 2);
-          code += "} from '" + include_name + ".js';\n";
-        }
-        code += "\n";
+
+      // re-export child namespace(s) in parent
+      const auto child_ns_level = it.second.ns->components.size() + 1;
+      for (const auto &it2 : ns_defs_) {
+        if (it2.second.ns->components.size() != child_ns_level) continue;
+        auto ts_file_path = it2.second.path + ".ts";
+        code += "export * as " + it2.second.symbolic_name + " from './";
+        code += it2.second.path + ".js';\n";
+        export_counter++;
       }
 
-      code += flat_file_;
-      const std::string filename =
-          GeneratedFileName(path_, file_name_, parser_.opts);
-      SaveFile(filename.c_str(), code, false);
-    } else {
-      for (auto it = imports_all_.begin(); it != imports_all_.end(); it++) {
-        code += it->second.export_statement + "\n";
+      if (export_counter > 0) SaveFile(it.second.filepath.c_str(), code, false);
+    }
+  }
+
+  bool generateBundle() {
+    if (parser_.opts.ts_flat_files) {
+      std::string inputpath;
+      std::string symbolic_name = file_name_;
+      if (parser_.current_namespace_->components.size() > 0) {
+        std::string path = namer_.Directories(*parser_.current_namespace_,
+                                              SkipDir::TrailingPathSeperator);
+        inputpath = path + ".ts";
+        symbolic_name = parser_.current_namespace_->components.back();
+      } else {
+        inputpath = path_ + file_name_ + ".ts";
       }
-      const std::string path =
+      std::string bundlepath =
           GeneratedFileName(path_, file_name_, parser_.opts);
-      SaveFile(path.c_str(), code, false);
+      bundlepath = bundlepath.substr(0, bundlepath.size() - 3) + ".js";
+      std::string cmd = "esbuild";
```

Comment:
@aardappel @bjornharrtell You are invoking the `esbuild` binary directly from `flatc` which I don't think we should do. I know it is convenient to do, but it just feels odd (and error prone) to assume the command exists.

I think it would be better to do the bundling external to flatc.

Reply:
It'd be nice to be able to inject the path to the binary through an environment or flag rather than assuming it is a particular name in the PATH.

Reply:
I agree it is perhaps a bit odd but it's convenient. If we make it entirely external it would require a fair bit of explanation/docs on how to use it.

Reply:
I'm hesistant to remove it.. I don't really see what the win is to externalise it even though it's a bit "exotic". But if really have to to get this PR going again I'll do it. In that case, should the suggested command line be output as instruction instead?

Reply:
This seems a bit too opaque to invoke it from `flatc`, and is a dependency we don't normally have. It would also need to be tested to be fully portable etc. I vote for not doing it. Doesn't seem that much harder to invoke it externally and it be so much cleaner.

Reply:
Ok, I've changed it to output the suggested command line instead of running it.

Remains to actually run those lines as part of souce generation in tests and I realize I'm not autotesting the bundled output. I'll see what I can do about that.

Reply:
I would rather have some external script (don't know what is common in JS/TS land) that executes `flatc` and does the bundling as two distinct steps. That way it can be documented with the script.